.github/workflows/CI.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  style:
+    name: Check Style
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Install Rust
+      run: rustup update beta && rustup default beta
+    - name: Install Rustfmt
+      run: rustup component add rustfmt
+    - name: Check formatting
+      run: cargo fmt --all -- --check
+
+  test:
+    name: Test
+    needs: [style]
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: -Dwarnings
+    strategy:
+      matrix:
+        rust:
+          - nightly
```

Comment:
GitHub Actions don't have allow_failures, so we probably need to use a way like https://stackoverflow.com/questions/57850553/github-actions-check-steps-status.